WcaOnRails/app/controllers/delegates_controller.rb

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+class DelegatesController < ApplicationController
+  before_action :authenticate_user!
```

Comment:
Are you saying there are other controllers where only some routes require login?

That would be a decent argument against the base class. Perhaps those controllers should be split, but perhaps this is not that important an distinction.